fastly/cloudfiles_test.go

```diff
@@ -126,80 +201,121 @@ func TestClient_Cloudfiles(t *testing.T) {
-		t.Errorf("bad placement: %q", cloudfiles.Placement)
+	if cloudfilesCreateResp1.PublicKey != cloudfilesGetResp.PublicKey {
+		t.Errorf("bad public_key: %q", cloudfilesCreateResp1.PublicKey)
 	}
-	if cloudfiles.PublicKey != ncloudfiles.PublicKey {
-		t.Errorf("bad public_key: %q", cloudfiles.PublicKey)
+	if cloudfilesCreateResp1.CompressionCodec != cloudfilesGetResp.CompressionCodec {
+		t.Errorf("bad compression_codec: %q", cloudfilesCreateResp1.CompressionCodec)
 	}
 
 	// Update
-	var ucloudfiles *Cloudfiles
+	var cloudfilesUpdateResp1, cloudfilesUpdateResp2, cloudfilesUpdateResp3 *Cloudfiles
 	record(t, "cloudfiles/update", func(c *Client) {
-		ucloudfiles, err = c.UpdateCloudfiles(&UpdateCloudfilesInput{
+		cloudfilesUpdateResp1, err = c.UpdateCloudfiles(&UpdateCloudfilesInput{
+			ServiceID:        testServiceID,
+			ServiceVersion:   tv.Number,
+			Name:             "test-cloudfiles",
+			NewName:          String("new-test-cloudfiles"),
+			User:             String("new-user"),
+			Period:           Uint(0),
+			FormatVersion:    Uint(2),
+			CompressionCodec: String("zstd"),
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	record(t, "cloudfiles/update2", func(c *Client) {
+		cloudfilesUpdateResp2, err = c.UpdateCloudfiles(&UpdateCloudfilesInput{
+			ServiceID:        testServiceID,
+			ServiceVersion:   tv.Number,
+			Name:             "test-cloudfiles-2",
+			CompressionCodec: String("zstd"),
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	record(t, "cloudfiles/update3", func(c *Client) {
+		cloudfilesUpdateResp3, err = c.UpdateCloudfiles(&UpdateCloudfilesInput{
 			ServiceID:      testServiceID,
 			ServiceVersion: tv.Number,
-			Name:           "test-cloudfiles",
-			NewName:        String("new-test-cloudfiles"),
-			User:           String("new-user"),
-			Period:         Uint(0),
-			GzipLevel:      Uint(0),
-			FormatVersion:  Uint(2),
+			Name:           "test-cloudfiles-3",
+			GzipLevel:      Uint(9),
 		})
 	})
 	if err != nil {
 		t.Fatal(err)
 	}
-	if ucloudfiles.Name != "new-test-cloudfiles" {
-		t.Errorf("bad name: %q", ucloudfiles.Name)
+
+	if cloudfilesUpdateResp1.Name != "new-test-cloudfiles" {
+		t.Errorf("bad name: %q", cloudfilesUpdateResp1.Name)
+	}
+	if cloudfilesUpdateResp1.User != "new-user" {
+		t.Errorf("bad user: %q", cloudfilesUpdateResp1.User)
+	}
+	if cloudfilesUpdateResp1.GzipLevel != 0 {
+		t.Errorf("bad gzip_level: %q", cloudfilesUpdateResp1.GzipLevel)
 	}
-	if ucloudfiles.User != "new-user" {
-		t.Errorf("bad user: %q", ucloudfiles.User)
+	if cloudfilesUpdateResp1.Period != 0 {
+		t.Errorf("bad period: %q", cloudfilesUpdateResp1.Period)
 	}
-	if ucloudfiles.GzipLevel != 0 {
-		t.Errorf("bad gzip_level: %q", ucloudfiles.GzipLevel)
+	if cloudfilesUpdateResp1.FormatVersion != 2 {
+		t.Errorf("bad format_version: %q", cloudfilesUpdateResp1.FormatVersion)
 	}
-	if ucloudfiles.Period != 0 {
-		t.Errorf("bad period: %q", ucloudfiles.Period)
+	if cloudfilesUpdateResp2.CompressionCodec != "zstd" {
```

Comment:
I mentioned this below in the other test file, but shouldn't we be checking `cloudfilesUpdateResp1` also is set to `zstd` for its compression? as it was `snappy` when initially created.

Reply:
Done